front/src/common/Map/WarpedMap/core/grids.tsx

```diff
@@ -0,0 +1,245 @@
+/* eslint-disable prefer-destructuring, no-plusplus */
+import { Feature, LineString, Position } from 'geojson';
+import { clamp, cloneDeep, keyBy, meanBy } from 'lodash';
+import length from '@turf/length';
+import center from '@turf/center';
+import { featureCollection, lineString, polygon } from '@turf/helpers';
+import destination from '@turf/destination';
+import bearing from '@turf/bearing';
+
+import {
+  GridFeature,
+  GridIndex,
+  Grids,
+  PointsGrid,
+  Triangle,
+} from 'common/Map/WarpedMap/core/helpers';
+import vec, { Vec2 } from 'common/Map/WarpedMap/core/vec-lib';
+import { PolygonZone } from 'types';
+
+/**
+ * Base helpers to manipulate grids:
+ */
+export function getGridIndex(grid: GridFeature): GridIndex {
+  return keyBy(grid.features, (feature) => feature.properties.triangleId);
+}
+
+export function featureToPointsGrid(grid: GridFeature, steps: number): PointsGrid {
+  const points: PointsGrid = [];
+  const gridIndex = getGridIndex(grid);
+  const stripsPerSide = grid.features.length / steps / 2 / 2;
+
+  for (let i = 0; i < steps; i++) {
+    points[i] = points[i] || {};
+    points[i + 1] = points[i + 1] || {};
+    for (let direction = -1; direction <= 1; direction += 2) {
+      for (let j = 0; j < stripsPerSide; j++) {
+        const inside = gridIndex[`step:${i}/strip:${direction * (j + 1)}/inside`];
+        const outside = gridIndex[`step:${i}/strip:${direction * (j + 1)}/outside`];
+        const [[p00, p10, p01]] = inside.geometry.coordinates;
+        const [[p11]] = outside.geometry.coordinates;
+
+        points[i][direction * j] = p00;
+        points[i][direction * (j + 1)] = p01;
+        points[i + 1][direction * j] = p10;
+        points[i + 1][direction * (j + 1)] = p11;
+      }
+    }
+  }
+
+  return points;
+}
+export function pointsGridToFeature(points: PointsGrid): GridFeature {
```

Comment:
No need to export this function (but you can add an empty line before it :) )
```suggestion

function pointsGridToFeature(points: PointsGrid): GridFeature {
```